tests/basic/clone/clone.c

```diff
@@ -0,0 +1,99 @@
+#include <errno.h>
+#include <stdarg.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <signal.h>
+
+#define _GNU_SOURCE
+#include <sched.h>
+#include <unistd.h>
+#include <sys/syscall.h>
+
+static char child_stack[8192];
+static char *child_stack_end = child_stack + 8192;
+static char child_tls[4069];
+
+volatile int thread_started;
+__attribute__((weak)) int lkl_syscall(int, long*);
+
+static void assert(int cond, const char *msg, ...)
+{
+	if (cond) return;
+	va_list ap;
+	va_start(ap, msg);
+	vfprintf(stderr, msg, ap);
+	va_end(ap);
+	fprintf(stderr, "\nTEST_FAILED\n");
+	exit(-1);
+}
+
+int do_clone(int syscall_number, long *args);
+static int clone_wrapper(int (*fn)(void *), void *child_stack, int flags, void *arg, pid_t *ptid, void *newtls, pid_t *ctid)
+{
+	long args[6] = 
+		{
+			(long)flags,
+			((long)child_stack) - 16,
+			(long)ptid,
+			(long)ctid,
+			(long)newtls,
+			0
+		};
+	void **stack = (void**)child_stack;
+	stack[-1] = fn;
+	stack[-2] = arg;
+	return do_clone(220, args);
+}
+
+static void sigsegv(int signo, siginfo_t *si, void *addr)
+{
+	fprintf(stderr, "Fault: %p (%p)\n", si->si_addr, addr);
+	_Exit(0);
```

Comment:
why 0 here?

Reply:
Dead code from when I was getting segfaults in the kernel because the stack was misaligned in the clone assembly wrapper.